monitoring/zabbix_link_template.py

```diff
@@ -0,0 +1,138 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from ansible.module_utils.basic import *
+try:
+    from pyzabbix import ZabbixAPI
+    HAS_ZABBIX = True
+except ImportError:
+    HAS_ZABBIX = False
+
+DOCUMENTATION = '''
+---
+module: zabbix_update
+short_description: Link/unlink given hosts list with given templates list
+description:
+  - Link/unlink given hosts list with given templates list
+
+version_added: "2.3"
+author: "Michal Mankowski (@MichalMankowski)"
+requirements:
+  - "python >= 2.6"
+  - "pyzabbix >= 0.7.2"
+options:
+  server_url:
+     description:
+         - Address of zabbix server
+     required: True
+  username:
+     description:
+         - name of user to login to zabbix
+     required: True
+  password:
+     description:
+         - password for user to login to zabbix
+     reuqierd: True
+  host:
+     description:
+         - list of hosts to update
+     required: True
+  template:
+     description:
+         - list of templates to link/unlink
+  state:
+     description:
+         - state present link templates, absent unlink templates
+     required: False
+'''
+
+EXAMPLES = '''
+---
+- name: link templates
+  local_action:
+    module: zabbix_update
+    server_url: "http://127.0.0.1"
+    username: "test"
+    password: "test"
+    template:
+    - "Template App HTTP Service"
+    host:
+    - "Zabbix server"
+    state: present
+'''
+RETURN = '''
+#defaults
+'''
+
+
+def link_templates(template_ids, host_ids, state, zapi, check_mode):
+    for host in host_ids:
+        for template in template_ids:
+            if state == "present" and not check_mode:
+                zapi.host.massadd(hosts=host, templates=template)
+            elif state == "absent" and not check_mode:
+                zapi.host.update(hostid=host, templates_clear=template)
+
+
+def get_resource_ids(zabbix_resources, filter_list, filter_field, result_field):
+    filtered_results = [item for item in zabbix_resources if item[filter_field] in filter_list]
+    diff = set(filter_list).difference(set([item[filter_field] for item in filtered_results]))
+    if len(diff):
+        raise Exception("Resources not found: {}".format(", ".join(diff)))
+    return [item[result_field] for item in filtered_results]
+
+
+def main():
+    fields = {
+        "server_url": {"required": True, "type": "str"},
+        "user": {"required": True, "type": "str"},
+        "password": {"required": True, "type": "str"},
+        "template": {"required": True, "type": "list"},
```

Comment:
As this is a list, it would make sense to add an alias for templates (plural) e.g. `"aliases": ['templates']` (also do not forget to add it to the docs)